archive_converter.py

```diff
@@ -13,8 +13,57 @@
 # limitations under the License.
 
 import email
+import os
 from setup_gmail import Message
 
+class ArchiveMessageIndex(object):
+    def __init__(self):
+        self._lookup_index = {} # Maps MessageKey to Thread
+        self._messages_seen = {} # Maps message.id to message
+
+    def find_thread(self, key: str) -> Message:
+        return self._lookup_index[key]
+
+    def update(self):
+        fully_updated = False
+        new_messages = []
+
+        for filename in os.listdir('test_data'):
+            if filename.endswith(".txt"):
+                email = generate_email_from_file('test_data/' + filename)
+                if(email.id not in self._messages_seen):
+                    new_messages.append(email)
+        self._add_messages_to_index(new_messages)
+
+    def _add_messages_to_index(self, new_messages):
```

Comment:
Mind adding a docstring for this function as well?

Would be good to document the pre/post-conditions of this func.
E.g. one of the key responsibilities is that it populates `Message.children`